examples/supply-chain-app/Dockerfile

```diff
@@ -0,0 +1,139 @@
+FROM node:12.20.1-alpine3.12 as builder
+
+RUN apk update
+
+# Need JDK for the openapi-generator that is part of the build process
+RUN apk add --no-cache openjdk8=8.252.09-r0
+
+# Need git because some of our npm depedencies might be coming
+# straight from github instead of being an npm package on npmjs.com.
+RUN apk add --no-cache git
+
+# Some install scripts of the npm package fabric-network need python
+RUN apk add --no-cache python3 py3-pip
+
+RUN npm install modclean -g
+
+WORKDIR /
+RUN mkdir /app/
+WORKDIR /app/
+COPY ./ ./
+RUN npm ci
+RUN ./node_modules/.bin/lerna clean --yes
+RUN ./node_modules/.bin/lerna bootstrap
+RUN npm run build:dev:backend
+RUN npm run webpack:dev:web
+RUN npm run build:dev:frontend -- --scope='@hyperledger/cactus-example-supply-chain-frontend'
+RUN ./node_modules/.bin/lerna clean --yes
+RUN ./node_modules/.bin/lerna bootstrap -- --production --no-optional
+
+# RUN modclean --run --patterns="default:*" --path ./examples/cactus-example-supply-chain-backend/
+# RUN modclean --run --patterns="default:*" --path ./examples/cactus-example-supply-chain-business-logic-plugin/
+# RUN modclean --run --patterns="default:*" --path ./examples/cactus-example-supply-chain-frontend/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-api-client/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-cmd-api-server/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-cockpit/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-common/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-core-api/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-core/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-plugin-consortium-manual/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-plugin-keychain-memory/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-plugin-keychain-vault/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-plugin-ledger-connector-besu/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-plugin-ledger-connector-fabric/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-plugin-ledger-connector-quorum/
+# RUN modclean --run --patterns="default:*" --path ./packages/cactus-test-tooling/
+
+RUN rm -rf ./packages/cactus-test-plugin*
+RUN rm -rf ./packages/cactus-test-cmd*
+RUN rm -rf ./packages/cactus-test-api*
+RUN rm -rf ./node_modules/
+
+FROM docker:20.10.2-dind-rootless as runner
+
+USER root
+
+RUN apk update
+
+# Add bash for convenience, sh has less features
+RUN apk add --no-cache bash=5.0.17-r0
+
+# Need curl for healthchecks
+RUN apk add --no-cache curl=7.69.1-r3
+
+# The file binary is used to inspect exectubles when debugging container image issues
+RUN apk add --no-cache file=5.38-r0
+
+RUN apk add --no-cache nodejs=12.20.1-r0
```

Comment:
It looks wierd to update nodejs and npm again.

Reply:
> It looks wierd to update nodejs and npm again.

@sfuji822 I agree. :-) Unfortunately the Alpine Linux packagers decided to package them separately for some reason so it's not actually installing npm twice because on Alpine Linux if you install the nodejs package then that will not give you npm out of the box like it does on other operating systems.